main.py

```diff
@@ -982,6 +992,28 @@ def _get_circuits_buffer(self):
                 self.circuits[c_id] = evc
         return self.circuits
 
+    @staticmethod
+    def check_disabled_component(uni_a: object, uni_z: object):
```

Comment:
Did you mean to set a `UNI` type hint here? `object` is too generic.

```suggestion
    def check_disabled_component(uni_a: UNI, uni_z: UNI):
```